manager/MaCh3Modes.cpp

```diff
@@ -0,0 +1,177 @@
+//MaCh3 includes
+#include "MaCh3Modes.h"
+
+// *******************
+MaCh3Modes::MaCh3Modes(std::string const &filename) {
+// *******************
+
+  // Load config
+  YAML::Node config = YAML::LoadFile(filename);
+
+  std::string GetMaCh3ModeName(const int Index);
+  NModes = 0;
+
+  Title = config["Title"].as<std::string>();
+  Generator = config["GeneratorName"].as<std::string>();
+
+  std::vector<std::string> names = config["MaCh3Modes"].as<std::vector<std::string>>();
+
+  for(size_t i = 0; i < names.size(); i++)
+  {
+    DeclareNewMode(names[i],
+                    config[names[i]]["Name"].as<std::string>(),
+                    config[names[i]]["GeneratorMaping"].as<std::vector<int>>());
+  }
+  // Add unknown category, it's better to have garbage category where all undefined modes will go rather than get random crashes
+  DeclareNewMode("UNKNOWN_BAD",
+                 "UNKNOWN_BAD",
+                 {});
+  // This is hack to not have bad mode
+  NModes--;
+
+  PrepareMap();
+
+  Print();
+}
+
+// *******************
+void MaCh3Modes::Print() {
+// *******************
+
+  MACH3LOG_INFO("Printing MaCh3 Modes Called: {}", Title);
+
+  MACH3LOG_INFO("========================================================================");
+  MACH3LOG_INFO("{:<5} {:2} {:<20} {:2} {:<20} {:2} {:<30}", "#", "|", "Name", "|", "FancyName", "|", std::string(Generator+" Modes"));
+  MACH3LOG_INFO("------------------------------------------------------------------------");
+  for(int i = 0; i < NModes; ++i) {
+    auto Name = fMode[i].Name;
+    auto FancyName = fMode[i].FancyName;
+    auto Values = fMode[i].GeneratorMaping;
+
+    std::string generatorModes;
+    for (const int& element : Values) {
+      generatorModes += std::to_string(element) + " ";
+    }
+    MACH3LOG_INFO("{:<5} {:2} {:<20} {:2} {:<20} {:2} {:<30}", i, "|", Name, "|", FancyName, "|", generatorModes);
+  }
+  MACH3LOG_INFO("========================================================================");
+
+  MACH3LOG_INFO("==========================");
+  MACH3LOG_INFO("{:<10} {:2} {:<30}", std::string(Generator + " Modes"), "|", "Name");
+  MACH3LOG_INFO("--------------------------");
+  for (size_t i = 0; i < ModeMap.size(); ++i) {
+    MACH3LOG_INFO("{:<10} {:2} {:<30}", i, "|", fMode[ModeMap[i]].Name);
+  }
+  MACH3LOG_INFO("==========================");
+}
+
+// *******************
+MaCh3Modes_t MaCh3Modes::EnsureModeNameRegistered(std::string const &name) {
+// *******************
+  if (Mode.count(name)) {
+    return Mode[name];
+  }
+  MaCh3Modes_t index = Mode.size();
+  Mode[name] = index;
+  return index;
+}
+
+// *******************
+void MaCh3Modes::DeclareNewMode(std::string const &name,
+                    std::string const &fancyname,
+                    std::vector<int> const &GenMap) {
+// *******************
+  ModeInfo newinfo;
```

Comment:
similarly, here if you have the struct just be mode then you have "Mode newMode;" and I think it's super clear this way? Again, very minor.